src/trace/trace.cpp

```diff
@@ -530,7 +583,7 @@ otf2::definition::region Trace::intern_region(const LineInfo& info)
 
 otf2::definition::string Trace::intern(const std::string& name)
 {
-    auto ref = strings_.size();
+    auto ref = string_ref();
     auto ret = strings_.emplace(std::piecewise_construct, std::forward_as_tuple(name),
```

Comment:
To avoid costly constructions of `definition::string` objects, this should be a `try_emplace`.